vm/ignite/src/main.rs

```diff
@@ -91,3 +92,496 @@ fn main() -> Result<()> {
+        loop x < 3 {
+            x = x + 1;
+            
+            if x == 2 {
+                break;
+              }
+        }
+        x
+        ";
+        test_pass(t, "2");
+
+        // sum of naturals - nested loop
+        let t = r"
+        let count = 0;
+        let x = 0;
+        let end = 10;
+        
+        loop x < end|| x == end{
+            let j = 0;
+            
+            loop j < x {
+                count = count + 1;
+                j = j + 1;
+            }
+            
+            x = x + 1;
+        }
+        
+        count
+        ";
+        test_pass(t, "55");
+
+        // nested, both have break - break targets correct loop each time
+        let t = r"
+        let count = 0;
+        let x = 0;
+        
+        loop x < 10 || x == 10 {
+            let j = 0;
+            
+            
+            loop {
+                    if !(j < x) {
+                          break;
+                    }
+                    
+                count = count + 1;
+                j = j + 1;
+                
+                if j == 5 {
+                    break;
+                }
+            }
+            
+            x = x + 1;
+            
+            if x == 7 {
+                break;
+            }
+        }
+        
+        count
+        ";
+        test_pass(t, "20");
+
+        // Triple nested
+        test_pass(
+            r"
+        let count = 0;
+
+        let x = 0;
+        loop x < 5 || x == 5 {
+            let y = 0;
+            
+            loop y < 5 || y == 5 {
+                let z = 0;
+                
+                loop {
+                        if !(z < 5 || z == 5) {
+                            break;
+                        }
+                        
+                    count = count + 1;
+                    z = z + 1;
+        
+                    if z == 3 {
+                        break;
+                    }
+                }
+                
+                y = y + 1;
+        
+                if y == 3 {
+                    break;
+                }
+            }
+            
+            x = x + 1;
+        
+            if x == 3 {
+                break;
+            }
+        }
+        
+        count
+        ",
+            "27",
+        );
+    }
+
+    #[test]
+    fn test_e2e_fib() {
+        // loop-fib-01.rst
+        let t = r"
+        let n : int = 10; // Calculate the 10th (0 idx) Fibonacci number = 55
+        let fib_prev : int = 0;
+        let fib_current : int = 1;
+        let fib_next : int = 0; 
+        
+        let i = 1; // Start from the 1st Fibonacci number
+        
+        loop i < n {
+            fib_next = fib_prev + fib_current;
+            fib_prev = fib_current;
+            fib_current = fib_next;
+            i = i + 1;
+        }
+        
+        if n == 0 {
+            fib_prev
+        } else {
+            fib_current
+        }
+        ";
+        test_pass(t, "55");
+    }
```

Comment:
Can refer to `vm/ignite/tests/vm_cli.rs` to test how to binaries interact directly. Lets not cluster the main.rs file